core/variant/variant_parser.cpp

```diff
@@ -1424,7 +1457,7 @@ Error VariantWriter::write(const Variant &p_variant, StoreStringFunc p_store_str
 		} break;
 		case Variant::FLOAT: {
 			String s = rtosfix(p_variant.operator double());
-			if (s != "inf" && s != "nan") {
+			if (s.find_char('n') == -1) { // ("inf", "inf_neg", "nan") all has 'n' and it's faster.
```

Comment:
I'm not sure I like this "optimization".

- There's potential for breakage if we add more identifiers with a "n" in their name.
- It's less explicit.
- Is it actually faster? I think that would depend on the length of `s`, as `find` needs to iterate over all characters *always*, while `!=` only needs to iterate characters as long as they match.

Reply:
Yup I didn't like that. Resolved in latest push.